chia/util/struct_stream.py

```diff
@@ -37,21 +37,24 @@ def parse_metadata_from_name(cls: Type[_T_StructStream]) -> Type[_T_StructStream
         raise ValueError(f"cls.BITS must be a multiple of 8: {cls.BITS}")
 
     if cls.SIGNED:
-        cls.MAXIMUM_EXCLUSIVE = 2 ** (cls.BITS - 1)
+        cls.MAXIMUM = 2 ** (cls.BITS - 1) - 1
         cls.MINIMUM = -(2 ** (cls.BITS - 1))
     else:
-        cls.MAXIMUM_EXCLUSIVE = 2**cls.BITS
+        cls.MAXIMUM = 2**cls.BITS - 1
         cls.MINIMUM = 0
 
+    cls.MINIMUM = cls(cls.MINIMUM)
+    cls.MAXIMUM = cls(cls.MAXIMUM)
+
     return cls
 
 
 class StructStream(int):
-    SIZE = 0
-    BITS = 0
-    SIGNED = False
-    MAXIMUM_EXCLUSIVE = 0
-    MINIMUM = 0
+    SIZE: int
+    BITS: int
+    SIGNED: bool
+    MAXIMUM: int
+    MINIMUM: int
```

Comment:
Why drop the values here?  They specifically made this type unusable directly.  Though I think having the values ends up incompatible with the above comment about making `StructStream` generic.  `main.py:8: error: ClassVar cannot contain type variables  [misc]`

Anyways, I think these are all class vars.  And we might want an alternative explicit mechanism for blocking instantiation depending on where all this ends up.  Could have a failing `.__init__()` here and assign the working one in `parse_metadata_from_name()` or somesuch.  Yeah, makes the name bad.  Just throwing out thoughts since I haven't actually tried to solve it all yet.

```suggestion
    SIZE: ClassVar[int]
    BITS: ClassVar[int]
    SIGNED: ClassVar[bool]
    MAXIMUM: ClassVar[int]
    MINIMUM: ClassVar[int]
```